hutch_python/plugins/happi.py

```diff
@@ -0,0 +1,51 @@
+"""
+This plugin depends on a JSON happi database stored somewhere in the file
+system. The path of this file is supplied in the YAML as ``filename``.
+Finally, if a smaller subsection of the databse is desired, a requirements
+specification can be used to limit the number of devices that are loaded.
+
+The plugin then handles; initalizing the ``happi.Client``, finding the
+containers that match the specified requirements, then using the device
+loading utilities from happi to create instantiated devices
+
+Example
+-------
+.. code:: YAML
+
+    happi:
+        filename: path/to/my_db.json
+        requirements:
+            active: True
+            beamline: MFX
```

Comment:
Are we only supporting one source file and one filter?

Reply:
I think the plan for February should be one JSON file and the Questionnaire. You can have as many filters as you like, but only one set. You can also not have a list of acceptable values for the reasons mentioned above.

Reply:
I see now. I feel like the plugin has limited use then, kind of just a "Get me my hutch stuff" button. That's still useful.

Reply:
Yeah unfortunately we are limited by the `happi` search. What is annoying is that MongoDb has a bunch of nice search features, but if we want `happi` to be backend agnostic it has to be able to everything it does for Mongo but with a JSON backend. That means writing the search code ourselves. It has been on the to-do list but I always wanted to wait for a need instead of a want. We can discuss if we need more complication than this. I really intended this as just a first pass M.V.P effort.